### PR TITLE
Publish symbols to both public and internal servers

### PR DESCRIPTION
**Pull request by DefaultRyan**
This moves the private unstripped symbols to the MSFT internal server, and produces stripped symbols to publish to the public symbol servers.

### CODE REVIEW

**Comment by riverar** [comment]
Doesn't this also strip private symbols from open source artifacts? Ideally full symbol information would be available for open bits.

**Comment by DefaultRyan** [comment]
> Doesn't this also strip private symbols from open source artifacts? Ideally full symbol information would be available for open bits.

That's a fair point. Perhaps stripping the symbols does more harm than good in this case. @jonwis, any thoughts?

**Comment by DrusTheAxe** [comment]
> That's a fair point. Perhaps stripping the symbols does more harm than good in this case.

It does. Please don't strip symbols from open source code. It degrades fidelity and productivity for no upside

**Comment by jonwis** [comment]
> That's a fair point. Perhaps stripping the symbols does more harm than good in this case. @jonwis, any thoughts?

Agreed. Don't strip from OSS components.

From our closed-source helpers I'm reasonably certain we should _not_ strip either. Nothing in there will be "secret sauce" - it's wrappers and helpers around internals that need a public surface.  (Ideally we have a mode of stripping that retains good method names for the top layer.)

File: build/ProjectReunion-BuildFoundation.yml [hunk 1]
```diff
@@ -232,7 +232,20 @@ jobs:
 #      dir /s $(Build.ArtifactStagingDirectory)\symbols
 
   - task: PublishSymbols@2
-    displayName: 'Publish symbols (without source indexing)'
+    displayName: 'Publish private symbols to internal server (without source indexing)'
+    inputs:
+      searchPattern: '$(Build.ArtifactStagingDirectory)\symbols\**\*.pdb'
+      symbolServerType: 'TeamServices'
+# this ADO task does not support indexing of github sources currently :-(
```
Comment (riverar) [inline build/ProjectReunion-BuildFoundation.yml:239]: Nit: Indent and capitalize

File: build/ProjectReunion-BuildFoundation.yml [hunk 2]
```diff
@@ -232,7 +232,20 @@ jobs:
...
-    displayName: 'Publish symbols (without source indexing)'
+    displayName: 'Publish private symbols to internal server (without source indexing)'
+    inputs:
+      searchPattern: '$(Build.ArtifactStagingDirectory)\symbols\**\*.pdb'
+      symbolServerType: 'TeamServices'
+# this ADO task does not support indexing of github sources currently :-(      
+      indexSources: false 
+      detailedLog: true
+    # There is a bug which causes this task to fail if LIB includes an inaccessible path (even though it does not depend on it).
+    # To work around this issue, we just force LIB to be any dir that we know exists.
```
Comment (riverar) [inline build/ProjectReunion-BuildFoundation.yml:243]: Nit: Is there a bug number tracking this? I'd add it if so. That would give folks a method of quickly checking up on the bug to answer the question "can I delete these lines now?"

File: build/StripSymbols.ps1 [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return Get-ItemProperty -Path $windowsSDKRegPath | Select-Object -ExpandProperty $windowsDebuggingToolsRegRootKey
+    }
+    catch
+    {
+        return $null
+    }
+}
+
+function Strip-PrivateInformation(
```
Comment (riverar) [inline build/StripSymbols.ps1:28]: Nit: `Strip` isn't an [approved PowerShell verb](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands).
```suggestion
function Remove-PrivateSymbolInformation(
```

**Review by riverar — APPROVED** [review]
Just several non-blocking nits. Thank you very much for accommodating the public symbol changes.

File: build/ProjectReunion-BuildFoundation.yml (hunk 2 above)
Comment (DefaultRyan) [reply build/ProjectReunion-BuildFoundation.yml:243]: Not that I'm aware of. This comment is actually just copy pasted from the lines above, and I'm not the original author of the other one. 🤷‍♂️

**Pull request merged**